services/management/commands/turku_service_import/services.py

```diff
@@ -0,0 +1,136 @@
+
+from services.management.commands.services_import.keyword import KeywordHandler
+from services.management.commands.turku_service_import.utils import set_tku_translated_field, set_field, \
+    get_turku_resource
+from services.models import ServiceNode, Service
+
+UTC_TIMEZONE = pytz.timezone('UTC')
+
+
+class ServiceImporter:
+    nodesyncher = ModelSyncher(ServiceNode.objects.all(), lambda obj: obj.ext_id)
+    servicesyncher = ModelSyncher(Service.objects.all(), lambda obj: obj.id)
+
+    def __init__(self, syncher=None, noop=False, logger=None, importer=None):
+        self.syncher = syncher
+        self.noop = noop
+        self.logger = logger
+        self.importer = importer
+
+    def import_services(self):
+        keyword_handler = KeywordHandler(logger=self.logger)
+        self._import_service_nodes(keyword_handler)
+        self._import_services(keyword_handler)
+
+    def _import_service_nodes(self, keyword_handler):
+        service_classes = get_turku_resource('palveluluokat')
+        tree = self._build_servicetree(service_classes)
+        for parent_node in tree:
+            latest_node_id = 0
+            latest_node = ServiceNode.objects.order_by('id').last()
+            if latest_node:
+                latest_node_id = latest_node.id
+            self._handle_service_node(parent_node, keyword_handler, latest_node_id)
+        self.nodesyncher.finish()
+
+    def _import_services(self, keyword_handler):
+        services = get_turku_resource('palvelut')
+        for service in services:
+            self._handle_service(service, keyword_handler)
+        self.servicesyncher.finish()
+
+    def _save_object(self, obj):
+        if obj._changed:
+            obj.last_modified_time = datetime.now(UTC_TIMEZONE)
+            obj.save()
+            if self.importer:
+                self.importer.services_changed = True
+
+    def _build_servicetree(self, service_classes):
+        tree = [s_cls for s_cls in service_classes if 'ylatason_koodi' not in s_cls]
+        for parent in tree:
+            self._add_service_tree_children(parent, service_classes)
+
+        return tree
+
+    def _add_service_tree_children(self, parent_classes, service_classes):
+        parent_classes['children'] = [s_cls for s_cls in service_classes if
+                                      s_cls.get('ylatason_koodi') == parent_classes['koodi']]
+
+        for child_ot in parent_classes['children']:
+            self._add_service_tree_children(child_ot, service_classes)
+
+    def _handle_service_node(self, node, keyword_handler, latest_node_id):
+        obj = self.nodesyncher.get(node['koodi'])
+        if not obj:
+            latest_node_id += 1
+            obj = ServiceNode(
+                id=latest_node_id,
+                ext_id=node['koodi']
+            )
+            obj._changed = True
+
+        obj.name = node['nimi']
+        if set_field(obj, 'name', node, 'nimi'):
+            obj._changed = True
```

Comment:
While doing it there is possible of course, it kinda don't like that the function would do two things at that point. Right not it is pretty clear what it does, it takes the node and sets the values that is in there to the values on the object. If `_change` would be set there as well, the it would start adding new fields. `set_field` would then no loger the a general purpose function that takes a dict, and sets its values to the corresponding attribute of an object. It would become `set_fields_on_these_specific_object_types` (not exactly the name I would pick but you get the point).

Reply:
Fair enough :) I still think the function should be then either renamed or a new function added, `set_syncher_object_field()` or something, to avoid having to duplicate the `_changed` update logic everywhere the function is used. Not a big deal of course, your PR, do as you please :)

Reply:
> Not a big deal of course, your PR, do as you please :)

PRs should not work this way 😄 I want to include code and we decide together if that code is worth including in its current state or not. Your point adding a new function is completely valid, I'll add that to reduce boiler plate.